addons/im_livechat/static/src/embed/core/history_service.js

```diff
@@ -0,0 +1,67 @@
+/* @odoo-module */
+
+import { registry } from "@web/core/registry";
+import { getCookie, setCookie } from 'web.utils.cookies';
+
+export class HistoryService {
+    static HISTORY_COOKIE = "im_livechat_history";
+    static HISTORY_LIMIT = 15;
+
+    constructor(env, services) {
+        this.rpc = services.rpc;
```

Comment:
`rpc` should be typed: `ReturnType<typeof import("@web/core/network/rpc_service").rpcService.start>`